### PR TITLE
Introduce PMacc back-end switch in cmake

### PR DESCRIPTION
**Pull request by erikzenker**
This PR introduces the property `PMACC_ACC_BACK_END` to the the PMacc cmake build suite.
This property is either `None`, `GPU Nvidia`, `CPU OMP2` or `CPU C++11 threads` and sets the correct alpaka macros. Based on this macros the correct accelerator, stream and device is choosen.

### CODE REVIEW

**Comment by erikzenker** [comment]
Note: The code currently only compiles for the `GPU Nvidia` back-end, the other back-ends throw compile time errors and, therefore, need to be fixed in a further PR.

File: src/libPMacc/include/types_alpaka.hpp [hunk 1]
```diff
@@ -57,13 +74,13 @@ namespace alpaka
     using Acc = ::alpaka::acc::AccCpuOmp2Threads<
         T_Dim,
         IdxSize
-    >;
-#else
+    >;    
+#elif (defined(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE))
```
Comment (erikzenker) [inline src/libPMacc/include/types_alpaka.hpp]: ..._ENABLED

File: src/libPMacc/include/types_alpaka.hpp [hunk 2]
```diff
@@ -39,7 +36,27 @@ namespace alpaka
...
+     */
+    template<
+        typename T_Dim
+    >
+    using Acc = ::alpaka::acc::AccGpuCudaRt<
+        T_Dim,
+        IdxSize
+    >;
+    
+#elif (defined(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE))
```
Comment (erikzenker) [inline src/libPMacc/include/types_alpaka.hpp]: ..._ENABLED

Comment (erikzenker) [inline src/libPMacc/include/types_alpaka.hpp]: Wrong accelerator needs to be : ALPAKA_ACC_CPU_B_SEQ_T_OMP2...

**Comment by erikzenker** [comment]
Update ...compiles and runs now with all available accs.

**Comment by bussmann** [comment]
Very cool!

**Comment by psychocoderHPC** [comment]
well done

**Pull request merged**